In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [9]:
import numpy as np
import pandas as pd
path = '/content/drive/MyDrive/Colab Notebooks/penguins.csv'
penguins = pd.read_csv(path)
print("penguins modülü yüklendi!")

penguins modülü yüklendi!


In [10]:
display(penguins.head())

,id,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex,year
0,0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,male,2007
1,1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,female,2007
2,2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,female,2007
3,3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN,2007
4,4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,female,2007


In [ ]:
print(penguins.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     342 non-null    float64
 4   bill_depth_mm      342 non-null    float64
 5   flipper_length_mm  342 non-null    float64
 6   body_mass_g        342 non-null    float64
 7   sex                333 non-null    object 
 8   year               344 non-null    int64  
dtypes: float64(4), int64(2), object(3)
memory usage: 24.3+ KB
None


**print(penguins.info())** kodu bize; 344 satır olduğunu, dataların kaç tane NAN' a sahip olduğunu ve data type'lerini göstermektedir

In [ ]:
print(penguins.describe())

               id  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      342.000000     342.000000         342.000000   
mean   171.500000       43.921930      17.151170         200.915205   
std     99.448479        5.459584       1.974793          14.061714   
min      0.000000       32.100000      13.100000         172.000000   
25%     85.750000       39.225000      15.600000         190.000000   
50%    171.500000       44.450000      17.300000         197.000000   
75%    257.250000       48.500000      18.700000         213.000000   
max    343.000000       59.600000      21.500000         231.000000   

       body_mass_g         year  
count   342.000000   344.000000  
mean   4201.754386  2008.029070  
std     801.954536     0.818356  
min    2700.000000  2007.000000  
25%    3550.000000  2007.000000  
50%    4050.000000  2008.000000  
75%    4750.000000  2009.000000  
max    6300.000000  2009.000000  


**describe** fonksiyonu bize sadece sayısal olan (int,float) sütunların sayısı,ortalaması,standart sapması, ortanca değeri, maximum değeri gibi bilgileri verir.

In [6]:
print(penguins.isnull().sum())

id                   0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64


In [5]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].median())

In [7]:
penguins['sex'] = penguins['sex'].map({'male': 0, 'female': 1})

In [11]:
features = penguins[['bill_length_mm', 'bill_depth_mm', 'flipper_length_mm', 'body_mass_g', 'island', 'sex', 'year']]
label = penguins['species']





In [12]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(features, label, test_size=0.2, random_state=42)

print("Eğitim ve Test Setinin Boyutları:")
print("X_train:", X_train.shape)
print("X_test:", X_test.shape)
print("y_train:", y_train.shape)
print("y_test:", y_test.shape)

Eğitim ve Test Setinin Boyutları:
X_train: (275, 7)
X_test: (69, 7)
y_train: (275,)
y_test: (69,)


x --> features
y --> label

**X_train:**

Modelin eğitilmesi (fit edilmesi) için kullanılan giriş verilerini (özellikleri) içerir.

**X_test:**

Model eğitildikten sonra, modelin performansını değerlendirmek için kullanılan giriş verilerini içerir.

**y_train:**

Modelin öğrenmesi gereken hedef değişkenin değerlerini içerir.

**y_test:**

Model eğitildikten sonra, modelin tahminlerini gerçek değerlerle karşılaştırmak için kullanılan hedef değişkenin değerlerini içerir.




 modelin gerçek dünyada ne kadar iyi çalıştığını ölçmek için onu hiç görmediği bir veriyle test etmeliyiz. İşte bu yüzden veri setini ikiye ayırıyoruz